poll/poll_linux.go

```diff
@@ -115,6 +118,8 @@ func (p *Poller) WaitWriteCtx(ctx context.Context) error {
 		return nil
```

Comment:
Is there the potential for a default case here? What happens if its not one of those two?